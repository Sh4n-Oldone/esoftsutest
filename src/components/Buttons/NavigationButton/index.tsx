import React from 'react';
import {StyleSheet, View} from 'react-native';
import {buttonsColor, mainTextColor} from '../../../constants/constants';
import TypicalButton from '../TypicalButton';

const NavigationButton: React.FC<{navigateCallback: () => void; text: string}> =
  ({navigateCallback, text}) => {
    return (
      <View style={styles.container}>
        <TypicalButton
          action={navigateCallback}
          text={text}
          buttonStyles={styles.button}
          textStyles={styles.buttonText}
        />
      </View>
    );
  };

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    position: 'absolute',
    bottom: 10,
  },
  button: {
    backgroundColor: buttonsColor,
    paddingVertical: 6,
    paddingHorizontal: 10,
    minWidth: 100,
    borderRadius: 40 / 2,
  },
  buttonText: {
    color: mainTextColor,
    textAlign: 'center',
  },
});

export default NavigationButton;
