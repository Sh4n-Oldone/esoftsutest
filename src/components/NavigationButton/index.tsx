import React from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {buttonsColor, mainTextColor} from '../../constants/constants';

const NavigationButton: React.FC<{navigateCallback: () => void; text: string}> =
  ({navigateCallback, text}) => {
    return (
      <View style={styles.container}>
        <TouchableOpacity style={styles.button} onPress={navigateCallback}>
          <Text style={styles.buttonText}>{text}</Text>
        </TouchableOpacity>
      </View>
    );
  };

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    position: 'absolute',
    bottom: 10,
  },
  button: {
    backgroundColor: buttonsColor,
    paddingVertical: 6,
    paddingHorizontal: 10,
    minWidth: 100,
    borderRadius: 40 / 2,
  },
  buttonText: {
    color: mainTextColor,
    textAlign: 'center',
  },
});

export default NavigationButton;
