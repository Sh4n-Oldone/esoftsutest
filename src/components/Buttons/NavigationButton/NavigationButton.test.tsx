import 'react-native';
import React from 'react';
import NavigationButton from './index';

import renderer from 'react-test-renderer';

describe('Correctly rendered button', () => {
  it('NavigationButton with 2 props renders correctly', () => {
    const btn = renderer
      .create(<NavigationButton navigateCallback={() => {}} text={'bttn'} />)
      .toJSON();
    expect(btn).toMatchSnapshot();
  });

  it('NavigationButton with 2 props have only child', () => {
    const btn = renderer
      .create(
        <NavigationButton navigateCallback={() => {}} text={'to search'} />,
      )
      .toJSON();
    expect(btn.children.length).toBe(1);
  });
});
