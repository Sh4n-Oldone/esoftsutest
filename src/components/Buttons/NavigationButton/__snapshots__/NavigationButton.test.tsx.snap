// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered button NavigationButton with 2 props renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 10,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "position": "absolute",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00659C",
        "borderRadius": 20,
        "minWidth": 100,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 6,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      bttn
    </Text>
  </View>
</View>
`;
